pkg/controller/kubelet-config/kubelet_config_bootstrap.go

```diff
@@ -0,0 +1,118 @@
+package kubeletconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// RunKubeletBootstrap generates MachineConfig objects for mcpPools that would have been generated by syncKubeletConfig
+func RunKubeletBootstrap(templateDir string, kubeletConfigs []*mcfgv1.KubeletConfig, controllerConfig *mcfgv1.ControllerConfig, features *configv1.FeatureGate, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	var res []*mcfgv1.MachineConfig
+	managedKeyExist := make(map[string]bool)
+	userDefinedSystemReserved := make(map[string]string)
+	// Validate the KubeletConfig CR if exists
+	for _, kubeletConfig := range kubeletConfigs {
+		if kubeletConfig == nil {
```

Comment:
I don't think this can be nil, so maybe this is redundant

Reply:
move the check before the for loop and do a if kubeletConfig != nil{

Reply:
Ah technically this gets checked outside of this function, because for this to get called, it must get past `if len(kconfigs) > 0` in the main bootstrap logic.

We can add an extra check in this function, but I think it should be fine to just remove completely (would just skip the for loop if by some miracle we got here with an empty KC anyways, so wouldn't error)